test/test_api/test_results_manager.py

```diff
@@ -0,0 +1,216 @@
+           f"\tOptimisation Metric: {api._metric.name}",
+           f"\tBest validation score: {max(search_results.opt_scores)}",
+           "\tNumber of target algorithm runs: 16", f"\tNumber of successful target algorithm runs: {n_success}",
+           f"\tNumber of crashed target algorithm runs: {n_crashed}",
+           f"\tNumber of target algorithms that exceeded the time limit: {n_timeout}",
+           f"\tNumber of target algorithms that exceeded the memory limit: {n_memoryout}"]
+
+    assert isinstance(api.sprint_statistics(), str)
+    assert all([m1 == m2 for m1, m2 in zip(api.sprint_statistics().split("\n"), msg)])
+
+
+@pytest.mark.parametrize('run_history', (None, RunHistory()))
+def test_check_run_history(run_history):
+    manager = ResultsManager()
+    manager.run_history = run_history
+
+    try:
+        manager._check_run_history()
+    except RuntimeError:
+        pass
+    else:
+        raise RuntimeError('The error was not properly catched.')
+
+
+T, NT = 'traditional', 'non-traditional'
+SCORES = [0.1 * (i + 1) for i in range(10)]
+
+
+@pytest.mark.parametrize('include_traditional', (True, False))
+@pytest.mark.parametrize('metric', (accuracy, log_loss))
+@pytest.mark.parametrize('origins', ([T] * 5 + [NT] * 5, [T, NT] * 5, [NT] * 5 + [T] * 5))
+@pytest.mark.parametrize('scores', (SCORES, SCORES[::-1]))
+def test_get_incumbent_results(include_traditional, metric, origins, scores):
+    manager = ResultsManager()
+    cs = ConfigurationSpace()
+    cs.add_hyperparameter(CSH.UniformFloatHyperparameter('a', lower=0, upper=1))
+
+    configs = [0.1 * (i + 1) for i in range(len(scores))]
+    if metric.name == "log_loss":
+        # This is to detect mis-computation in reversion
+        metric._optimum = 0.1
+
+    best_cost, best_idx = np.inf, -1
+    for idx, (a, origin, score) in enumerate(zip(configs, origins, scores)):
+        config = Configuration(cs, {'a': a})
+
+        # conversion defined in:
+        # autoPyTorch/pipeline/components/training/metrics/utils.py::calculate_loss
+        cost = metric._optimum - metric._sign * score
+        manager.run_history.add(
+            config=config,
+            cost=cost,
+            time=1.0,
+            status=StatusType.SUCCESS,
+            additional_info={'opt_loss': {metric.name: score},
+                             'configuration_origin': origin}
+        )
+        if cost > best_cost:
+            continue
+
+        if include_traditional:
+            best_cost, best_idx = cost, idx
+        elif origin != T:
+            best_cost, best_idx = cost, idx
+
+    incumbent_config, incumbent_results = manager.get_incumbent_results(
+        metric=metric,
+        include_traditional=include_traditional
+    )
+
+    assert isinstance(incumbent_config, Configuration)
+    assert isinstance(incumbent_results, dict)
+    best_score, best_a = scores[best_idx], configs[best_idx]
+    assert abs(best_score - cost2metric(best_cost, metric)) < 1e-5
```

Comment:
do you think we can use [`np.isclose`](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html`)